fv3_cap.F90

```diff
@@ -1212,7 +1212,7 @@ subroutine fv3_checkimport(gcomp, rc)
     type(ESMF_Clock)           :: clock
     type(ESMF_Time)            :: currTime, invalidTime
     type(ESMF_State)           :: importState
-    logical                    :: timeCheck1,timeCheck2
+    logical                    :: isValid
```

Comment:
This local variable `isValid` is used at three places and have three different meanings at all those places. I find it very confusing.
First it is used to indicate whether a field's timestamp is "a valid timestamp", from NUOPC's perspective, and this is the only place this variable should be used with this name. 
Second it is reused to check whether a field is not at "invalid time", where an invalid time is defined locally and set to an arbitrary/special value (year  99999999). Why is this necessary? How is this value (99999999) determined?
And finally `isValid` is reused for a third time to check whether a field is at current time. 
If all these checks are necessary, then three logical variables with different names should be used. For example  `isValid`, `isNotSpecialValue` and `isAtCurrentTime`, which will improve readability of the code at expense of adding two logical variables (8 bytes). 
I wonder if all these check can be combined in just one check that checks whether or not field's time is at current time?

Reply:
We could use multiple logical variables if needed for clarity. We could also write a single logical function combining all the tests for increased modularity. I don't believe such a function is required elsewhere in the code base, though, and it could lead to confusion in cases where a field timestamp has not been set at all.